airflow/providers/openlineage/utils/utils.py

```diff
@@ -73,6 +78,16 @@ def is_operator_disabled(operator: BaseOperator | MappedOperator) -> bool:
     return get_fully_qualified_class_name(operator) in conf.disabled_operators()
 
 
+def is_selective_lineage_enabled(obj) -> bool:
+    """If selective enable is active check if DAG or Task is enabled to emit events."""
+    if not conf.selective_enable():
+        return True
+    if isinstance(obj, DAG):
+        return is_dag_lineage_enabled(obj)
+    else:
+        return is_task_lineage_enabled(obj)
```

Comment:
Let’s add a guard so this is only called on operator objects. The error message would be confusing if this function is (incorrectly) used on a foreign object in the current implementation.

Reply:
Added.